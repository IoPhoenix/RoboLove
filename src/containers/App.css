@font-face {
    font-family: 'MrRobot';
    font-style: normal;
    font-weight: normal;
    src: local('MrRobot'), url('MRROBOT.woff') format('woff');
}

* {
    margin: 0;
    padding: 0;
}

h1 {
    margin-bottom: 1rem;
    color: #73f79c;
    font-weight: 200;
    font-family: 'MrRobot';
}