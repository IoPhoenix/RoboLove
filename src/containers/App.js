import React, {Component} from 'react';
import './App.css';
import Sticky from '../components/Sticky';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';

class App extends Component {
    constructor() {
        super()
        this.state = {
            robots: [],
            searchfield: ''
        }
    }

    componentDidMount() {
        fetch('https://jsonplaceholder.typicode.com/users')
        .then(response => response.json())
        .then(users => this.setState({robots: users}) );
    }

    onSearchChange = (e) => {
        this.setState({ searchfield: e.target.value });
    }

   render() {
    const { robots, searchfield } = this.state;
    const filteredRobots = robots.filter(robot => {
        return robot.name
                .toLowerCase()
                .includes(searchfield.toLowerCase());
    });

    return !robots.length
        ? <h1 className='tc ma5'>Loading...</h1>
        : (
            <div className='tc'>
                <h1 className='f1'>RoboLove</h1>
                <Sticky>
                    <SearchBox searchChange={this.onSearchChange} />
                </Sticky>                    
                <CardList className='flex justify-center' robots={filteredRobots} />                
            </div>
        )
    }
}

export default App;