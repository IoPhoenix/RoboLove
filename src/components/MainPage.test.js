import { shallow } from 'enzyme';
import React from 'react';
import App from '../containers/App';

it('expect to render App component', () => {
    expect(shallow(<App />)).toM atchSnapshot();
});

it('correctly increments the counter', () => {
    const mockStore = {
        robots: [],
        searchField: ''
    }
    const wrapper = shallow(<App store={mockStore} />);

    // expect(wrapper).toMatchSnapshot();
});

