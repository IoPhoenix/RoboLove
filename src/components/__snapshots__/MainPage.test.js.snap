// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    isPending={false}
    onRequestRobots={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    robots={Array []}
    searchField=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="f1"
        >
          RoboLove
        </h1>,
        <Sticky>
          <SearchBox />
        </Sticky>,
        <ErrorBoundary>
          <CardList
            className="flex justify-center"
            robots={Array []}
          />
        </ErrorBoundary>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "RoboLove",
          "className": "f1",
        },
        "ref": null,
        "rendered": "RoboLove",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <SearchBox />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchChange": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <CardList
            className="flex justify-center"
            robots={Array []}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "flex justify-center",
            "robots": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="f1"
          >
            RoboLove
          </h1>,
          <Sticky>
            <SearchBox />
          </Sticky>,
          <ErrorBoundary>
            <CardList
              className="flex justify-center"
              robots={Array []}
            />
          </ErrorBoundary>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "RoboLove",
            "className": "f1",
          },
          "ref": null,
          "rendered": "RoboLove",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <SearchBox />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "searchChange": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CardList
              className="flex justify-center"
              robots={Array []}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "flex justify-center",
              "robots": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
